typescript/web/src/components/datasets/dataset-list.context.tsx

```diff
@@ -0,0 +1,108 @@
+import { gql, useQuery } from "@apollo/client";
+import type { Dataset as DatasetType } from "@labelflow/graphql-types";
+import { isNil } from "lodash/fp";
+import { createContext, PropsWithChildren, useContext } from "react";
+import { usePagination } from "../pagination";
+
+export const getPaginatedDatasetsQuery = gql`
+  query getPaginatedDatasets(
+    $where: DatasetWhereInput
+    $first: Int!
+    $skip: Int!
+  ) {
+    datasets(where: $where, first: $first, skip: $skip) {
+      id
+      name
+      slug
+      images(first: 1) {
+        id
+        url
+        thumbnail500Url
+      }
+      imagesAggregates {
+        totalCount
+      }
+      labelsAggregates {
+        totalCount
+      }
+      labelClassesAggregates {
+        totalCount
+      }
+    }
+  }
+`;
+
+type UseQueryParamSetter = (
+  value: string,
+  updateType?: "replace" | "push" | "replaceIn" | "pushIn" | undefined
```

Comment:
Isn't it an existing type in the library?

Maybe that some of the utility types listed in this doc could help you: https://www.typescriptlang.org/docs/handbook/utility-types.html

Reply:
No it's not an existing type but I can use a exported one for the different strings

Reply:
I've seen you've made to code to use [`UrlUpdateType`](https://github.com/pbeshai/use-query-params/blob/master/packages/use-query-params/src/types.ts#L11) now. That's what I've asked for actually, perfect :)
The sentence about the utility types was just in case the type was only declared as an argument of the query-params setter function (i.e. `setParam(........, updateType: "replace" | "push" | "replaceIn" | "pushIn" | undefined)` which would have forced you to use something like `Parameters<typeof setParam>[1]`.